addons/l10n_ar_base/models/res_partner.py

```diff
@@ -19,17 +16,59 @@ class ResPartner(models.Model):
     l10n_ar_id_number = fields.Char(
         string='Identification Number',
     )
-    l10n_ar_id_category_id = fields.Many2one(
-        string="Identification Category",
-        comodel_name='l10n_ar_id_category',
+    l10n_ar_identification_type_id = fields.Many2one(
+        string="Identification Type",
+        comodel_name='l10n_ar.identification.type',
         index=True,
         auto_join=True,
     )
+    same_id_number_partner = fields.Html(
+        string='Partner with same Identification Number',
+        compute='_compute_same_same_id_number_partner',
+        store=False,
+    )
+    l10n_ar_country_code = fields.Char(
+        related="company_id.l10n_ar_country_code",
+    )
+
+    @api.depends('l10n_ar_id_number', 'l10n_ar_identification_type_id')
+    def _compute_same_same_id_number_partner(self):
+        for partner in self:
+            partner_id = partner.id
+            partner_id_number = partner.l10n_ar_id_number
+            partner_id_type = partner.l10n_ar_identification_type_id
+            if isinstance(partner_id, models.NewId):
+                # deal with onchange(), which is always called on a single
+                # record
+                partner_id = self._origin.id
+            domain = [
+                ('l10n_ar_id_number', '=', partner_id_number),
+                ('l10n_ar_identification_type_id', '=', partner_id_type.id)]
+            if partner_id:
+                related_partners = partner.search([
+                    '|', ('id', 'parent_of', partner_id),
+                    ('id', 'child_of', partner_id)])
+                domain += [('id', 'not in', related_partners.ids)]
```

Comment:
@jco-odoo indeed: that's the reason why this method is different, I try to used as the original same_vat_partner but do not work because the "and not partner.parent_id" the mentioned logic was not identifying repeated identification numbers in contacts that have same parent or even any parent.

> "children" need to same identification as "parents"?

The children does not need to have the same identification of the parents. Example:

 * Parent has CUIT 123456
     * Child 2 has DNI 98765 (diff from parent) 

> Can not we skip this if the identification type is CUIT?

I will check this with the functional team and let you know soon if we can or not skip the CUIT case, As far as I remember there are cases when we like to have the children contacts with a CUIT defined different from the parent's CUIT